File: pueblo-golf-tournament-api/Entities/Division.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Entities
{
    public class Division : BaseEntity
    {
        public string? Name { get; set; }
        public long TournamentId { get; set; }

        
        public virtual Tournament? Tournament { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/models/division.dart
import 'package:json_annotation/json_annotation.dart';

import 'base-model.dart';
part 'division.g.dart';

@JsonSerializable()
class Division extends BaseModel {
  final String name;
  final double tournamentId;

  Division({required this.name, required this.tournamentId, required super.id});

  factory Division.fromJson(Map<String, dynamic> json) =>
      _$DivisionFromJson(json);

  Map<String, dynamic> toJson() => _$DivisionToJson(this);
}
